Isolate key guide + tidying up (#40)
vegaprotocol/documentation: docs/tools/vega-wallet/cli-wallet/latest/guides/isolate-keys.md

---
sidebar_position: 4
title: Isolate keys
hide_title: false
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

If you want to move a key pair from a standard wallet into a wallet that can only sign and verify transactions, use the `key isolate` command. Isolating a key pair will extract a chosen key pair from one wallet and create an isolated wallet with only that key pair.

This optional step creates an extra layer of security, particularly for validators and others who are concerned about their wallets being compromised. 

> An isolated wallet can only contain a single key pair that has been stripped from its cryptographic node. A wallet without a cryptographic node, which is used to create keys, minimises the impact of that wallet being compromised, as it makes it impossible to retrieve or generate keys. 

Reviewer: ```suggestion
An isolated wallet can only contain a single key pair that has been stripped from its cryptographic node. A wallet without a cryptographic node minimises the damages if the wallet has been compromised, as it makes it impossible to generate keys or retrieve any other keys than isolated one. 
```

It's very badly worded, but I think sentence is too heavy and convoluted. We should find a better way to explain why not having this node is beneficial to security.

Author: An isolated wallet can only contain a single key pair that has been stripped from its cryptographic node. If a wallet without a cryptographic node is compromised, the damage is minimised as it is impossible to generate keys or retrieve any keys other than the isolated one. 

How about that ^ @ValentinTrinque?

Reviewer: It's honey to my ears.

<TabItem value="windows" label="Windows">

```bash
vega genesis new validator --country="XX" --info-url="VALIDATOR_URL" --name="NODE_NAME"
```

</TabItem>
<TabItem value="mac" label="MacOS">

```bash
./vega genesis new validator --country="XX" 
/ --info-url="VALIDATOR_URL" --name="NODE_NAME"
```

</TabItem>
<TabItem value="linux" label="Linux">

```bash
./vega genesis new validator --country="XX" 
/ --info-url="VALIDATOR_URL" --name="NODE_NAME"
```

</TabItem>
</Tabs>

### 3. Add details to genesis file

Add the following information to the genesis file under `validators` key:
```
{
  "address": "TENDERMINT_ADDRESS",
  "pub_key": {
    "type": "tendermint/PubKeyEd25519",
    "value": "TENDERMINT_PUBLIC_KEY"
  },
  "power": "10",
  "name": ""
}
```

Add the following information into the genesis file under `app_state.validators` key:

```
{
    "TENDERMINT_PUBLIC_KEY": {
      "id": "VALIDATOR_ID",
      "vega_pub_key": "ISOLATED_PUBLIC_KEY",
      "ethereum_address": "MY_ETHEREUM_ADDRESS",
      "tm_pub_key": "TENDERMINT_PUBLIC_KEY",
      "info_url": "VALIDATOR_URL",
      "country": "XX",
      "name": "NODE_NAME",
      "avatar_url": ""
    }
  }

> ```

Author: Thought you might say that. I put them together as a temporary option, to get it out without having to create a whole new section with just that one guide, as it's a little bit out of context without more guides. I'm willing to break them up if you think it's too confusing or irrelevant. Though... it will also take a little longer to get this one out if so :)

> ```

Reviewer: Alright, then it's fine.
